numpy/lib/tests/test_io.py

```diff
@@ -1918,6 +1918,15 @@ def test_single_dtype_w_implicit_names(self):
                         dtype=[(_, float) for _ in "abc"])
         assert_equal(mtest, ctrl)
 
+    def test_single_col_w_implicit_names(self):
+        # Test single column w implicit names
+        data = 'a_row_to_skip\na_header\n5\n6'
+        mtest = np.genfromtxt(TextIO(data), skip_header=1, delimiter=",", names=True, usecols=0)
+        # Note that squeeze doesn't work when specifying names
+        ctrl = np.array([(5,), (6,)],
+                        dtype=[('a_header', int)])
```

Comment:
Without commenting on the nature of the regression, this passed for me locally with this minor adjustment:

```diff
         ctrl = np.array([(5,), (6,)],
-                        dtype=[('a_header', int)])
+                        dtype=[('a_header', float)])
         assert_equal(mtest, ctrl)
```

Reply:
Yes, sorry the correction you have applied is needed. Actually the test passes without the change that I have made. But at line 1971
`converters = [StringConverter(dtype, locked=True`, parameter `dtype` has the header name in it and therefore the StringConverter does not find the right type and the converter[0].type is void, whereas it should be `float`. This only happens when you have a single column. While for multiple columns of data the `dtype_flat` parameter is passed in, which has the header names removed.
The correction I have provided makes it consistent with the multiple column approach and means the data type is found within the StringConverter object.

Reply:
I think this test is incorrect: note that the default value for `dtype` is float, so you should expect a float type here.

The dtype-inference-machinery is triggered with `dtype=None`, which gives the expected result:

```python
>>> np.genfromtxt(
...     StringIO(data), skip_header=1, delimiter=",", names=True, usecols=0, dtype=None
... )
array([(5,), (6,)], dtype=[('a_header', '<i8')])
```

AFAICT the regression has been fixed and the current behavior is correct --- I think this can be closed (unless I'm missing something @millen1m )